### PR TITLE
depends: add backports 11986 12402 13543

### PR DESCRIPTION
**Pull request by 10xcryptodev**
bitcoin#11986 - zeromq 4.2.3
bitcoin#12402 - expat 2.2.5, ccache 3.4.1, miniupnpc 2.0.20180203
bitcoin#13543 - Add RISC-V support

main updates on config.sub and config.guess and also is possible `make clean` on depends now

### CODE REVIEW

File: depends/packages/miniupnpc.mk [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
 
 define $(package)_set_vars
 $(package)_build_opts=CC="$($(package)_cc)"
 $(package)_build_opts_darwin=OS=Darwin LIBTOOL="$($(package)_libtool)"
 $(package)_build_opts_mingw32=-f Makefile.mingw
 $(package)_build_env+=CFLAGS="$($(package)_cflags) $($(package)_cppflags)" AR="$($(package)_ar)"
+$(package)_build_env+=CFLAGS=-D_DARWIN_C_SOURCE
 endef
 
 define $(package)_preprocess_cmds
```
Comment (PastaPastaPasta) [inline depends/packages/miniupnpc.mk:15]: missing changes in native_ccache.mk

Comment (10xcryptodev) [reply depends/packages/miniupnpc.mk:15]: @PastaPastaPasta Dash doesn't have this package and was also removed from bitcoin later

**Comment by 10xcryptodev** [comment]
@PastaPastaPasta done :+1:

**Comment by PastaPastaPasta** [comment]
Primary thought process imo was that I'm trying to minimize the number of PRs in 16 and get us into a feature freeze :joy: 

However, justification is that any of the fixes in these PRs where considered small enough that bitcoin didn't backport these to 0.16.x (which they had time to do based on when these were merged and when 0.16.3 was released) and in theory could introduce problems which would cause more testing / fixing 16...

I don't care that much, I'm just trying to limit the scope of 16 at this point, and this PR leans into 0.17 backports

**Comment by UdjinM6** [comment]
Might also need #13689 and #12466 as followups.

>bitcoin didn't backport these to 0.16.x

Indeed, no such backports here https://github.com/bitcoin/bitcoin/commits/0.16/depends. Ok, let's postpone it till 0.17 just in case.

**Pull request merged**